packages/gguf/src/types.ts

```diff
@@ -75,21 +86,40 @@ interface Tokenizer {
 	"tokenizer.ggml.bos_token_id": number;
 	"tokenizer.ggml.eos_token_id": number;
 	"tokenizer.ggml.add_bos_token": boolean;
-	"tokenizer.chat_template": string;
+	"tokenizer.chat_template"?: string;
 }
+interface NoTokenizer {
+	"tokenizer.ggml.model"?: undefined;
+}
+
+/// Models outside of llama.cpp: "rwkv" and "whisper"
 
-export type RWKV = ModelBase<"rwkv"> & { "rwkv.architecture_version": number };
-export type LLM = TransformerLLM | RWKV;
-export type Whisper = ModelBase<"encoder.whisper"> & ModelBase<"decoder.whisper">;
-export type Model = (LLM | Whisper) & Partial<Tokenizer>;
+export type RWKV = GGUFGeneralInfo<"rwkv"> &
+	ModelBase<"rwkv"> & {
+		"rwkv.architecture_version": number;
+	};
 
-export type GGUFMetadata = {
+// TODO: whisper.cpp doesn't yet support gguf. This maybe changed in the future.
+export type Whisper = GGUFGeneralInfo<"whisper"> &
+	ModelBase<"encoder.whisper"> &
+	ModelBase<"decoder.whisper"> & {
+		"whisper.encoder.mels_count": number;
+		"whisper.encoder.attention.head_count": number;
+		"whisper.decoder.attention.head_count": number;
+	};
+
+/// Types for parse output
+
+export type GGUFStrictType = true;
+export type GGUFNonStrictType = false;
+
+export type GGUFMetadata<T extends GGUFStrictType | GGUFNonStrictType = GGUFStrictType> = {
```

Comment:
small nit. Like in `TArchitecture`, we can use more descriptive generic name other than just `T`. Maybe `TGgufType` ?

Reply:
Implemented in https://github.com/huggingface/huggingface.js/pull/655/commits/c2afbdc080980c81ad37a933dfcea840dde9d3bc